lib/jsdom/living/helpers/focusing.js

```diff
@@ -102,3 +102,29 @@ exports.fireFocusEventWithTargetAdjustment = (name, target, relatedTarget, { bub
 
   target._dispatch(event);
 };
+
+exports.outsideUnopenedDialog = child => {
+  if (!child.parentNode) {
+    return true;
+  }
+
+  if (child.parentNode.tagName === "DIALOG" && !child.parentNode.open) {
+    return false;
+  }
+
+  return exports.outsideUnopenedDialog(child.parentNode);
+};
+
+exports.runAutoFocus = target => {
```

Comment:
Hmm. The way this is done is very unlike the spec at https://html.spec.whatwg.org/#flush-autofocus-candidates , including when you call it. Do you think it'd be feasible to implement the spec here? I'm worried that this kind of implementation will cause people to raise more bugs about autofocus not working as it does in browsers, instead of our current status of it just not being implemented...